src/fs/dir.rs

```diff
@@ -129,11 +129,31 @@ impl<'dir, 'ig> Files<'dir, 'ig> {
                     }
                 }
 
-                return Some(File::from_args(path.clone(), self.dir, filename, self.deref_links)
-                                 .map_err(|e| (path.clone(), e)))
+                let file = File::from_args(
+                    path.clone(),
+                    self.dir,
+                    filename,
+                    self.deref_links
+                ).map_err(|e| (path.clone(), e));
+
+                // Windows has it's own concept of hidden files, with a flag
+                // stored in the file's attributes
+                #[cfg(windows)]
+                match file {
+                    Err(err) => return Some(Err(err)),
+                    Ok(file) => {
+                        if !self.dotfiles && file.attributes().hidden {
+                            continue;
+                        }
+
+                        return Some(Ok(file));
+                    },
+                }
```

Comment:
Seems strange to unpack the file result and repack.  Could you do something like
```rust
                #[cfg(windows)]
                if !self.dotfiles && file.is_ok_and(|f| f.attributes().hidden) {
                    continue
                }
                return Some(file);
```

Reply:
Thankyou for this, I'm pretty new to Rust so I knew there was probably a better way to handle it

Reply:
So that approach moves the file & stops it returning below, this seems to work though:
```rust
#cfg(windows)]
if !self.dotfiles && file.as_ref().is_ok_and(|f| f.attributes().hidden) {
    continue;
}
return Some(file);